pkg/restapi/operation/didexchange/didexchange.go

```diff
@@ -350,21 +352,62 @@ func (c *Operation) startClientEventListener() error {
 		return fmt.Errorf("didexchange message event registration failed: %w", err)
 	}
 
-	// auto execute the actions
+	// event listeners
 	go func() {
-		err := service.AutoExecuteActionEvent(c.actionCh)
-		if err != nil {
-			logger.Errorf("auto action event execution failed: %s", err)
+		for {
+			select {
+			case e := <-c.actionCh:
+				c.handleActionEvents(e)
+			case e := <-c.msgCh:
+				c.handleMessageEvents(e)
+			}
 		}
 	}()
 
-	go func() {
-		for e := range c.msgCh {
-			// TODO https://github.com/hyperledger/aries-framework-go/issues/551 - Integrate Message event with Webhook
-			// for now, log the messages
-			logger.Infof("message event received : type=%s", e.Type)
+	return nil
+}
+
+func (c *Operation) handleMessageEvents(e service.StateMsg) {
```

Comment:
Consider returning the error so you can test this function?

Reply:
well, the agent can't meaningfully handle these other than logging as these are just stream events. 

But added return and logged by the caller.

Reply:
Yeah, I encountered this issue a bunch when I was unit testing. Returning an error at least allows me to unit test it, and in the non-test files I just logged. Not sure if there's a better way.